Upgrading to Typescript 4.4 results in error TS2345 when using web-streams-polyfill

I get error like this:

error TS2345: Argument of type 'ReadableStream<any>' is not assignable to parameter of type 'string | ReadableStream<any> | Uint8Array'.
  Type 'import("/home/bjorn/code/flatgeobuf/node_modules/web-streams-polyfill/dist/types/ts3.6/polyfill").ReadableStream<any>' is not assignable to type 'ReadableStream<any>'.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,10 @@
 > - 🏠 Internal
 > - 💅 Polish
 
+## Unreleased
+
+* 🐛 Fix compatibility with TypeScript's DOM types for `ReadableStream` and `TransformStream` ([#85](https://github.com/MattiasBuelens/web-streams-polyfill/issues/85), [#86](https://github.com/MattiasBuelens/web-streams-polyfill/pull/86))
+
 ## v3.1.0 (2021-07-21)
 
 * 🚀 Calling `ReadableStream.tee()` on a readable byte stream now returns two readable byte streams. ([#81](https://github.com/MattiasBuelens/web-streams-polyfill/pull/81))

--- a/etc/web-streams-polyfill.api.md
+++ b/etc/web-streams-polyfill.api.md
@@ -54,14 +54,14 @@ export class ReadableStream<R = any> {
         size?: undefined;
     });
     constructor(underlyingSource?: UnderlyingSource<R>, strategy?: QueuingStrategy<R>);
-    cancel(reason?: any): Promise<undefined>;
+    cancel(reason?: any): Promise<void>;
     getReader({ mode }: {
         mode: 'byob';
     }): ReadableStreamBYOBReader;
     getReader(): ReadableStreamDefaultReader<R>;
     get locked(): boolean;
     pipeThrough<T>(transform: ReadableWritablePair<T, R>, options?: StreamPipeOptions): ReadableStream<T>;
-    pipeTo(destination: WritableStream<R>, options?: StreamPipeOptions): Promise<undefined>;
+    pipeTo(destination: WritableStream<R>, options?: StreamPipeOptions): Promise<void>;
     tee(): [ReadableStream<R>, ReadableStream<R>];
     values(options?: ReadableStreamIteratorOptions): ReadableStreamAsyncIterator<R>;
 }
@@ -78,7 +78,7 @@ export interface ReadableStreamAsyncIterator<R> extends AsyncIterator<R> {
 export class ReadableStreamBYOBReader {
     // Warning: (ae-forgotten-export) The symbol "ReadableByteStream" needs to be exported by the entry point polyfill.d.ts
     constructor(stream: ReadableByteStream);
-    cancel(reason?: any): Promise<undefined>;
+    cancel(reason?: any): Promise<void>;
     get closed(): Promise<undefined>;
     read<T extends ArrayBufferView>(view: T): Promise<ReadableStreamBYOBReadResult<T>>;
     releaseLock(): void;
@@ -111,7 +111,7 @@ export class ReadableStreamDefaultController<R> {
 // @public
 export class ReadableStreamDefaultReader<R = any> {
     constructor(stream: ReadableStream<R>);
-    cancel(reason?: any): Promise<undefined>;
+    cancel(reason?: any): Promise<void>;
     get closed(): Promise<undefined>;
     read(): Promise<ReadableStreamDefaultReadResult<R>>;
     releaseLock(): void;
@@ -241,7 +241,7 @@ export type UnderlyingSourceStartCallback<R> = (controller: ReadableStreamDefaul
 // @public
 export class WritableStream<W = any> {
     constructor(underlyingSink?: UnderlyingSink<W>, strategy?: QueuingStrategy<W>);
-    abort(reason?: any): Promise<undefined>;
+    abort(reason?: any): Promise<void>;
     close(): Promise<undefined>;
     getWriter(): WritableStreamDefaultWriter<W>;
     get locked(): boolean;
@@ -257,13 +257,13 @@ export class WritableStreamDefaultController<W = any> {
 // @public
 export class WritableStreamDefaultWriter<W = any> {
     constructor(stream: WritableStream<W>);
-    abort(reason?: any): Promise<undefined>;
-    close(): Promise<undefined>;
+    abort(reason?: any): Promise<void>;
+    close(): Promise<void>;
     get closed(): Promise<undefined>;
     get desiredSize(): number | null;
     get ready(): Promise<undefined>;
     releaseLock(): void;
-    write(chunk: W): Promise<undefined>;
+    write(chunk: W): Promise<void>;
 }
 
 

--- a/etc/web-streams-polyfill.api.md
+++ b/etc/web-streams-polyfill.api.md
@@ -69,7 +69,7 @@ export class ReadableStream<R = any> {
 // @public
 export interface ReadableStreamAsyncIterator<R> extends AsyncIterator<R> {
     // (undocumented)
-    next(): Promise<IteratorResult<R>>;
+    next(): Promise<IteratorResult<R, undefined>>;
     // (undocumented)
     return(value?: any): Promise<IteratorResult<any>>;
 }
@@ -123,7 +123,7 @@ export type ReadableStreamDefaultReadResult<T> = {
     value: T;
 } | {
     done: true;
-    value: undefined;
+    value?: undefined;
 };
 
 // @public

--- a/src/lib/readable-stream.ts
+++ b/src/lib/readable-stream.ts
@@ -140,7 +140,7 @@ export class ReadableStream<R = any> {
    * The supplied `reason` argument will be given to the underlying source's {@link UnderlyingSource.cancel | cancel()}
    * method, which might or might not use it.
    */
-  cancel(reason: any = undefined): Promise<undefined> {
+  cancel(reason: any = undefined): Promise<void> {
     if (!IsReadableStream(this)) {
       return promiseRejectedWith(streamBrandCheckException('cancel'));
     }
@@ -229,9 +229,9 @@ export class ReadableStream<R = any> {
    *
    * Piping a stream will lock it for the duration of the pipe, preventing any other consumer from acquiring a reader.
    */
-  pipeTo(destination: WritableStream<R>, options?: StreamPipeOptions): Promise<undefined>;
+  pipeTo(destination: WritableStream<R>, options?: StreamPipeOptions): Promise<void>;
   pipeTo(destination: WritableStream<R> | null | undefined,
-         rawOptions: StreamPipeOptions | null | undefined = {}): Promise<undefined> {
+         rawOptions: StreamPipeOptions | null | undefined = {}): Promise<void> {
     if (!IsReadableStream(this)) {
       return promiseRejectedWith(streamBrandCheckException('pipeTo'));
     }

--- a/src/lib/readable-stream/async-iterator.ts
+++ b/src/lib/readable-stream/async-iterator.ts
@@ -30,7 +30,7 @@ import {
  * @public
  */
 export interface ReadableStreamAsyncIterator<R> extends AsyncIterator<R> {
-  next(): Promise<IteratorResult<R>>;
+  next(): Promise<IteratorResult<R, undefined>>;
 
   return(value?: any): Promise<IteratorResult<any>>;
 }
@@ -129,9 +129,9 @@ declare class ReadableStreamAsyncIteratorInstance<R> implements ReadableStreamAs
   /** @interal */
   _asyncIteratorImpl: ReadableStreamAsyncIteratorImpl<R>;
 
-  next(): Promise<ReadableStreamDefaultReadResult<R>>;
+  next(): Promise<IteratorResult<R, undefined>>;
 
-  return(value?: any): Promise<ReadableStreamDefaultReadResult<any>>;
+  return(value?: any): Promise<IteratorResult<any>>;
 }
 
 const ReadableStreamAsyncIteratorPrototype: ReadableStreamAsyncIteratorInstance<any> = {

--- a/src/lib/readable-stream/byob-reader.ts
+++ b/src/lib/readable-stream/byob-reader.ts
@@ -140,7 +140,7 @@ export class ReadableStreamBYOBReader {
   /**
    * If the reader is active, behaves the same as {@link ReadableStream.cancel | stream.cancel(reason)}.
    */
-  cancel(reason: any = undefined): Promise<undefined> {
+  cancel(reason: any = undefined): Promise<void> {
     if (!IsReadableStreamBYOBReader(this)) {
       return promiseRejectedWith(byobReaderBrandCheckException('cancel'));
     }

--- a/src/lib/readable-stream/default-reader.ts
+++ b/src/lib/readable-stream/default-reader.ts
@@ -23,7 +23,7 @@ export type ReadableStreamDefaultReadResult<T> = {
   value: T;
 } | {
   done: true;
-  value: undefined;
+  value?: undefined;
 }
 
 // Abstract operations for the ReadableStream.

--- a/src/lib/readable-stream/default-reader.ts
+++ b/src/lib/readable-stream/default-reader.ts
@@ -128,7 +128,7 @@ export class ReadableStreamDefaultReader<R = any> {
   /**
    * If the reader is active, behaves the same as {@link ReadableStream.cancel | stream.cancel(reason)}.
    */
-  cancel(reason: any = undefined): Promise<undefined> {
+  cancel(reason: any = undefined): Promise<void> {
     if (!IsReadableStreamDefaultReader(this)) {
       return promiseRejectedWith(defaultReaderBrandCheckException('cancel'));
     }

--- a/src/lib/writable-stream.ts
+++ b/src/lib/writable-stream.ts
@@ -123,7 +123,7 @@ class WritableStream<W = any> {
    * that there was an error doing so. Additionally, it will reject with a `TypeError` (without attempting to cancel
    * the stream) if the stream is currently locked.
    */
-  abort(reason: any = undefined): Promise<undefined> {
+  abort(reason: any = undefined): Promise<void> {
     if (!IsWritableStream(this)) {
       return promiseRejectedWith(streamBrandCheckException('abort'));
     }
@@ -688,7 +688,7 @@ export class WritableStreamDefaultWriter<W = any> {
   /**
    * If the reader is active, behaves the same as {@link WritableStream.abort | stream.abort(reason)}.
    */
-  abort(reason: any = undefined): Promise<undefined> {
+  abort(reason: any = undefined): Promise<void> {
     if (!IsWritableStreamDefaultWriter(this)) {
       return promiseRejectedWith(defaultWriterBrandCheckException('abort'));
     }
@@ -703,7 +703,7 @@ export class WritableStreamDefaultWriter<W = any> {
   /**
    * If the reader is active, behaves the same as {@link WritableStream.close | stream.close()}.
    */
-  close(): Promise<undefined> {
+  close(): Promise<void> {
     if (!IsWritableStreamDefaultWriter(this)) {
       return promiseRejectedWith(defaultWriterBrandCheckException('close'));
     }
@@ -756,8 +756,8 @@ export class WritableStreamDefaultWriter<W = any> {
    * Note that what "success" means is up to the underlying sink; it might indicate simply that the chunk has been
    * accepted, and not necessarily that it is safely saved to its ultimate destination.
    */
-  write(chunk: W): Promise<undefined>;
-  write(chunk: W = undefined!): Promise<undefined> {
+  write(chunk: W): Promise<void>;
+  write(chunk: W = undefined!): Promise<void> {
     if (!IsWritableStreamDefaultWriter(this)) {
       return promiseRejectedWith(defaultWriterBrandCheckException('write'));
     }

--- a/test/types/readable-stream.ts
+++ b/test/types/readable-stream.ts
@@ -123,8 +123,8 @@ const abortSignal: polyfill.AbortSignal = new AbortController().signal;
 const domUnderlyingSource: UnderlyingSource<string> = underlyingSource;
 // const domUnderlyingByteSource: UnderlyingByteSource = underlyingByteSource;
 
-// const domReadableStream: ReadableStream<string> = readableStream;
-// const domReadableByteStream: ReadableStream<Uint8Array> = readableByteStream;
+const domReadableStream: ReadableStream<string> = readableStream;
+const domReadableByteStream: ReadableStream<Uint8Array> = readableByteStream;
 
 const domDefaultController: ReadableStreamDefaultController<string> = defaultController;
 // const domByteStreamController: ReadableByteStreamController = byteStreamController;

--- a/test/types/transform-stream.ts
+++ b/test/types/transform-stream.ts
@@ -27,5 +27,5 @@ const readable: polyfill.ReadableStream<number> = transformStream.readable;
 
 // Compatibility with stream types from DOM
 const domTransformer: Transformer<string, number> = transformer;
-// const domTransformStream: TransformStream<string, number> = transformStream;
+const domTransformStream: TransformStream<string, number> = transformStream;
 const domController: TransformStreamDefaultController<number> = controller;

